core/src/main/java/com/backblaze/b2/json/B2JsonByteArrayOutputStream.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2021, Backblaze Inc. All Rights Reserved.
```

Comment:
This code looks remarkably similar to the code in ByteArrayOutputStream.java that i see in the IDE.  Similar enough that I don't think we actually wrote it and we need to figure something out.  Where did the code come from and what did we change?  (And why do we think that's reasonable?)

Reply:
Ops, not intentionally, I'd have to take that default copyright off. It is virtually a copy of standard version of ByteArrayOutputStream (minor changes of a couple of methods: grow and hugeCapacity). The thinking was: we were using the standard version of ByteArrayOutputStream and we could have an almost identical version of it with changed way of detecting the array allocation overflow and throwing IOException (as opposed OOM error). Somehow along the way the max array size (Integer.MAX_VALUE - 8) seems to be too big. Need to find a better max size limit.